ably-ios/ARTEventEmitter.h

```diff
@@ -9,14 +9,102 @@
 #import <Foundation/Foundation.h>
 #import "ARTTypes.h"
 
-@protocol ARTSubscription;
-
 @class ARTRealtime;
 
-@interface ARTEventEmitter : NSObject
+ART_ASSUME_NONNULL_BEGIN
+
+@interface __GENERIC(ARTEventListener, ItemType) : NSObject
+
+- (void)call:(ItemType)argument;
+
+@end
+
+@interface __GENERIC(ARTEventEmitter, EventType, ItemType) : NSObject
+
+- (__GENERIC(ARTEventListener, ItemType) *)on:(EventType)event call:(void (^)(ItemType))cb;
+- (void)on:(EventType)event callListener:(__GENERIC(ARTEventListener, ItemType) *)listener;
 
-- (instancetype)initWithRealtime:(ARTRealtime *)realtime;
-- (id<ARTSubscription>)on:(ARTRealtimeConnectionStateCb)cb;
-- (void)removeEvents;
+- (__GENERIC(ARTEventListener, ItemType) *)onAll:(void (^)(ItemType))cb;
```

Comment:
Based on our previous discussions about method overloading, I thought we had preferred to use overloading in the end.  If so, why do we have ` onAll` as opposed to `on(void (^)(ItemType))cb`?